src/Merchants/Content/Merchant/Services/RegistrationService.php

```diff
@@ -0,0 +1,110 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Production\Merchants\Content\Merchant\Services;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Checkout\Customer\SalesChannel\AccountRegistrationService as CustomerRegistrationService;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\Framework\Validation\DataBag\DataBag;
+use Shopware\Core\Framework\Validation\DataBag\RequestDataBag;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+
+class RegistrationService
+{
+    /**
+     * @var CustomerRegistrationService
+     */
+    private $customerRegistrationService;
+    /**
+     * @var Connection
+     */
+    private $connection;
+    /**
+     * @var EntityRepository
+     */
+    private $entityRepository;
+
+    public function __construct(
+        CustomerRegistrationService $customerRegistrationService,
+        Connection $connection,
+        EntityRepository $entityRepository
+    )
+    {
+        $this->customerRegistrationService = $customerRegistrationService;
+        $this->connection = $connection;
+        $this->entityRepository = $entityRepository;
+    }
+
+    public function registerMerchant(RequestDataBag $requestDataBag, SalesChannelContext $salesChannelContext): string {
+        $requestDataWithDefaultValues = $this->createDefaultDataForMerchant($requestDataBag);
+
+        $customerId = $this->customerRegistrationService->register($requestDataWithDefaultValues, false, $salesChannelContext);
+
+        $parameters = $requestDataBag->all();
+        $parameters['id'] = Uuid::randomHex();
+        $parameters['customerId'] = $customerId;
+        $parameters['salesChannelId'] = $salesChannelContext->getSalesChannel()->getId();
```

Comment:
the sales channel needs to be selected by the merchant and submitted through the api. Only the customer mus be scoped in the default (API) sales channel